src/chain/mod.rs

```diff
@@ -128,6 +128,8 @@ pub(crate) enum ChainSource {
 		server_url: String,
 		sync_config: ElectrumSyncConfig,
 		electrum_runtime_client: RwLock<Option<ElectrumRuntimeClient>>,
+		registered_txs_cache: Mutex<Vec<(Txid, ScriptBuf)>>,
+		registered_outputs_cache: Mutex<Vec<WatchedOutput>>,
```

Comment:
The naming `cache` confused me a bit for the pre-start txs/outs. Perhaps there is a better name.

Reply:
Something like queued, buffered or pending would - for me - match better what it is. But I guess that it depends on the definition of cache.